src/main/java/io/xpring/payid/PayIDComponents.java

```diff
@@ -0,0 +1,20 @@
+package io.xpring.payid;
+
+import org.immutables.value.Value;
+
+/**
+ * Represents components of a PayID.
+ */
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
+@Value.Immutable
+public interface PayIDComponents {
```

Comment:
You should just call this `PayId`

Reply:
talked offline - using components for now.